pkg/backend/openshiftcluster.go

```diff
@@ -89,13 +89,16 @@ func (ocb *openShiftClusterBackend) handle(ctx context.Context, log *logrus.Entr
 	case api.ProvisioningStateCreating:
 		log.Print("creating")
 
-		err = m.Create(ctx)
+		ok, err := m.Create(ctx)
 		if err != nil {
 			log.Error(err)
 			return ocb.endLease(ctx, stop, doc, api.ProvisioningStateFailed)
 		}
-
-		return ocb.endLease(ctx, stop, doc, api.ProvisioningStateSucceeded)
+		if ok {
+			return ocb.endLease(ctx, stop, doc, api.ProvisioningStateSucceeded)
+		}
+		stop()
```

Comment:
this will cause a 60 second delay because the lease is not formally given up, it just runs out.
can endLease (and updateAsyncOperation) be updated to handle a call that looks like `return ocb.endLease(ctx, stop, doc, api.ProvisioningStateCreating)`?  If possible it might help leaking special cases.